tests/e2e/conftest.py

```diff
@@ -155,18 +161,32 @@ def protostar(
         file.truncate()
         file.write(tomli_w.dumps(pyproject))
 
-    def _protostar(args: List[str], ignore_exit_code=False) -> str:
-        return (
-            run(
-                [path.join(tmp_path, "dist", "protostar", "protostar")] + args,
-                stdout=PIPE,
-                stderr=STDOUT,
-                check=(not ignore_exit_code),
-            )
-            .stdout.decode("utf-8")
-            .strip()
+    def _protostar(
+        args: List[str],
+        expect_exit_code: int = 0,
+        ignore_exit_code: bool = False,
+    ) -> str:
+        completed = run(
+            [path.join(tmp_path, "dist", "protostar", "protostar")] + args,
+            stdout=PIPE,
+            stderr=STDOUT,
+            encoding="utf-8",
         )
 
+        if not ignore_exit_code:
+            if completed.returncode != expect_exit_code:
```

Comment:
i'd write a custom assertion here, while returning some typed object. assertion could even be added as a method on that object, that way test would be more readable and "conventional". not needed ASAP though, can we add a task to the backlog for this?

Reply:
Discussed internally.